### PR TITLE
Afform Gui - Add support for entityRef fields

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Adds a widget for EntityRef to autocomplete contacts & other entities. Allows it to be changed to Number (for entering ID) or Select (for choosing another entity on a form).

Before
----------
Fields that pick other entities (e.g. contact reference fields like "Activity Created by") are not supported by Afform.

After
----------
EntityRef fields supported.

Comments
-----------
With this PR, creating an activity is finally functional. When you add one to the form, you'll see on the left side under "Values" the "Created by" field is pre-filled with "Current User". You can change it to an existing contact in the database, or have it populate from a contact on the form.

The Form Layout also supports EntityRef fields. By default they are an autocomplete to pick existing contacts, etc. from the database. You can also change the widget from EntityRef to Number or Select to either enter an id or select an entity on the form.

### CODE REVIEW

File: ext/afform/admin/afformEntities/Activity.php [hunk 1]
```diff
@@ -1,7 +1,13 @@
 <?php
 return [
   'entity' => 'Activity',
-  'defaults' => "{'url-autofill': '1'}",
+  'defaults' => "{
+    data: {
+      source_contact_id: 'user_contact_id',
```
Comment (seamuslee001) [inline ext/afform/admin/afformEntities/Activity.php:6]: @colemanw does source_contact_id show up in the UI? just wondering if this is changeable and what would user_contact_id reference if its an anon submission?

Comment (colemanw) [reply ext/afform/admin/afformEntities/Activity.php:6]: Yes this is the default and can be changed, but I think it's a sensible default. It makes adding a new activity look like this:
image

Comment (colemanw) [reply ext/afform/admin/afformEntities/Activity.php:6]: (also see **Comments** in PR description about this)

**Comment by seamuslee001** [comment]
@colemanw can you rebase please

**Pull request merged**